src/nix-build/nix-build.cc

```diff
@@ -476,7 +476,7 @@ static void main_nix_build(int argc, char * * argv)
         auto env = getEnv();
 
         auto tmp = getEnv("TMPDIR");
-        if (!tmp) tmp = getEnv("XDG_RUNTIME_DIR").value_or("/tmp");
+        if (!tmp || tmp->empty()) tmp = getEnv("XDG_RUNTIME_DIR").value_or("/tmp");
```

Comment:
Maybe not quite as likely, but should this also get some similar treatment? Maybe a helper function that is `getEnvOrDefault("the env var name", "the default value", bool default_if_empty = true)` that `defaultTempDir` could be implemented in terms of? (Or maybe this is premature optimization.)

Reply:
Very good point. I didn't expect to find `getEnvNonEmpty`, but there it was!